src/_pytest/main.py

```diff
@@ -503,7 +505,7 @@ def _collect(self, arg):
                             root = self._node_cache[pkginit]
                         else:
                             col = root._collectfile(pkginit)
-                            if col:
+                            if col and isinstance(col, Package):
```

Comment:
this one is part of creating headaches - in future we def need to sort that out

Reply:
Sure, I'm hoping @jonozzz can come up with a good solution to this. :grin:

Reply:
Totally agree, this needs a major rework, at least the code that went into main.py.